main.go

```diff
@@ -16,116 +14,121 @@ var (
-		FlagSet:     rootFlagSet,
-		Subcommands: []*ffcli.Command{listCmd, showCmd, connectCmd},
-		Exec: func(ctx context.Context, args []string) error {
-			// Get theme path from flag or environment variable.
-			themePath := *theme
-			if themePath == "" {
-				themePath = os.Getenv("WIFITUI_THEME")
-			}
+// We need a global backend to be accessible by the command handlers.
+var b wifi.Backend
 
-			if themePath != "" {
-				f, err := os.Open(themePath)
-				if err != nil {
-					return fmt.Errorf("failed to open theme file: %w", err)
-				}
-				defer f.Close()
-				loadedTheme, err := tui.LoadTheme(f)
-				if err != nil {
-					return fmt.Errorf("failed to load theme: %w", err)
-				}
-				tui.CurrentTheme = loadedTheme
-			}
-			return runTUI(b)
-		},
-	}
+// Execute is the handler for the "list" subcommand
+func (c *ListCommand) Execute(args []string) error {
+	return runList(os.Stdout, c.JSON, b)
+}
 
-	if err := root.Parse(os.Args[1:]); err != nil {
-		fmt.Fprintf(os.Stderr, "error: %v\n", err)
-		os.Exit(1)
+// Execute is the handler for the "show" subcommand
+func (c *ShowCommand) Execute(args []string) error {
+	return runShow(os.Stdout, c.JSON, c.Args.SSID, b)
+}
+
+// Execute is the handler for the "connect" subcommand
+func (c *ConnectCommand) Execute(args []string) error {
+	var security wifi.SecurityType
+	switch c.Security {
+	case "open":
+		security = wifi.SecurityOpen
+	case "wep":
+		security = wifi.SecurityWEP
+	case "wpa":
+		security = wifi.SecurityWPA
+	default:
+		return fmt.Errorf("invalid security type: %s", c.Security)
 	}
+	return runConnect(os.Stdout, c.Args.SSID, c.Passphrase, security, c.Hidden, b)
+}
+
+// main is the entry point of the application
+func main() {
+	var opts Options
+	parser := flags.NewParser(&opts, flags.Default)
+	parser.ShortDescription = "A simple TUI for managing wifi connections."
+	parser.LongDescription = "wifitui is a TUI and CLI for managing wifi connections."
 
-	if *version {
-		fmt.Println(Version)
-		os.Exit(0)
+	// Manually check for --version flag before parsing to avoid unnecessary backend init.
+	for _, arg := range os.Args[1:] {
+		if arg == "--version" {
+			fmt.Println(Version)
+			os.Exit(0)
+		}
 	}
 
+	// Initialize the backend before parsing, so it's available to Execute methods.
+	var err error
 	b, err = GetBackend()
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "error: %v\n", err)
 		os.Exit(1)
 	}
 
-	if err := root.Run(context.Background()); err != nil {
-		fmt.Fprintf(os.Stderr, "error: %v\n", err)
+	// Parse arguments. If a subcommand is found, its Execute method will be called.
+	_, err = parser.Parse()
+	if err != nil {
+		if flagsErr, ok := err.(*flags.Error); ok && flagsErr.Type == flags.ErrHelp {
+			// Help was requested, output is already printed.
+			os.Exit(0)
+		}
+		// Any other error is printed by go-flags.
 		os.Exit(1)
 	}
+
+	// If no subcommand was specified (parser.Active is nil), run the TUI.
+	if parser.Active == nil {
```

Comment:
This does not work. If I run `wifitui` without a subcommand, I get: `Please specify one command of: connect, list or show`

Let's add a `tui` subcommand and make sure we can run the parser without a subcommand to default to the `tui` subcommand.